### PR TITLE
[EGNR-179] Add CancelController

### PR DESCRIPTION
**Pull request by bautrukevich**

### CODE REVIEW

File: src/CancelController.ts [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+export interface CancelControllerInterface {
+  cancel(): void;
+  onCancel(fn: Function): void;
+}
 
-class CancelController {
-  private axiosCancel: CancelTokenSource
+class CancelController implements CancelControllerInterface {
+  private promise: Promise<void>
+  private resolve: (value?: void | PromiseLike<void> | undefined) => void = () => {}
```
Comment (jeetiss) [inline src/CancelController.ts]: We don't need value here

File: src/CancelController.ts [hunk 2]
```diff
@@ -1,23 +1,22 @@
-import {CancelTokenSource} from 'axios'
-const {CancelToken} = require('axios')
+export interface CancelControllerInterface {
```
Comment (jeetiss) [inline src/CancelController.ts]: Why do you add Interface here? Why do you export it? We don't have plan to provide tools for create another CancelController in user land

Comment (bautrukevich) [reply src/CancelController.ts]: But you want to pass CancelController as param and you should provide typing for this.

File: src/CancelController.ts (hunk 1 above)
Comment (bautrukevich) [reply src/CancelController.ts]: Because:

image

File: src/CancelController.ts (hunk 2 above)
Comment (bautrukevich) [reply src/CancelController.ts]: And you will use this CancelController in your code and you can see available methods

Comment (jeetiss) [reply src/CancelController.ts]: I use CancelController as type

**Pull request merged**